### PR TITLE
Decrease thumbnail upstream timeout and add timing logs for thumbnail requests

### LINKED ISSUE #4570: Investigate and decrease image proxy upstream request timeout
## Problem

Image proxy requests time out at 15 seconds. Similar to #4507, thumbnail requests are highly susceptible to upstream provider problems. If Flickr has a bad day, we'll have a bad day because we can't generate thumbnails (or validate dead links, to tie it back to that example).

## Description

We should perform a similar analysis as I did for #4507. Add logging to the thumbnail requests to time how long they take so we can analyse them. Log the time, status, url, and maybe go ahead and log the `provider` slug specifically so we can more easily segment the analysis along those lines (I wish we had done so for the dead link validation).

Deploy the logs for a week and follow the same approach to analysing the times as I took for the link validation. Find out how long (average, p95, p99) it takes for successful thumbnails to come back. Find out if there are meaningful differences between providers that we could modulate the timeout based on. etc etc.

The goal is to figure out if we can lower the 15-second timeout without creating significant negative consequences. Some failed thumbnails must be accepted, but something to consider is whether if a request times out (specifically times out) if we could try increasing the timeout the next time that thumbnail gets requested, and do so progressively until we hit some maximum (whether that's 15 seconds or otherwise).

## Additional context

This is mostly a way to investigate whether thumbnail timeouts really need to be so high. If they do, that's fine, but it'd be good to know.

It is not, however, likely to be an alternative to segmenting our API request timing monitors, as thumbnail requests will probably always take a little longer than our other requests, and along with search, will always be the most important to know without having the speedy outliers or volume of one or the other affect the visibility of problems with the other.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #4570 by @sarayourfriend

## Description

This PR decreases the thumbnail upstream request timeout as discussed in the linked issue. It also decreases the timeout for the HEAD request for the extension to the same default value. Finally, it adds timing logs for both these requests so we can further evaluate appropriate timeouts for each. For example, hopefully we can decrease the HEAD request timeout significantly (like 0.5 seconds at least). But, because the analysis in the linked issue was done with timings of the entire thumbnail request, it's not possible to know which part of the request actually takes 4 seconds. Therefore, the total timeout of outbound requests is now 8 seconds, which is still way lower than the 25 total seconds and still a good improvement in line with the goals of the issue.

To facilitate request timing in any circumstance we might want it, I've created an [aiohttp `TraceConfig` to log timing with context variables we've used so far](https://docs.aiohttp.org/en/stable/client_advanced.html#client-tracing). This only affects requests that configure themselves to activate the timing logs, as verified in the unit tests for the new trace config. This generalises the functionality added in #4508 for dead link checks, and therefore is able to replace that original implementation. As such, you'll see the code to manually time those requests is gone now, replaced by the trace config version. Because of that, I've also removed the separate timeout status we were caching before. This is safe for existing data because we check for specific _live_ status codes when using the cached data, and never actually referred to the difference between the timeout and general error statuses returned from `_head` except when inserting them into the cache.

## Testing Instructions

All existing unit tests must continue to pass without modification.

Run the API locally with `ov just api/up`. Open the logs with `ov just logs web` and make thumbnail requests and see the timing logs are present. Follow the testing instructions from #4508 and confirm they still work. The logs produced in this PR should be identical to the ones described in the linked PR, including the same contextual information, as well as `provider`, which was added later in #4598.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/compose.yml [hunk 1]
```diff
@@ -53,3 +53,12 @@ services:
...
       - web
+
+  httpbin:
+    profiles:
+      - api
+    # Use go-httpbin because it's only 15 MB compared to upstream HTTPBin's 204 MB
+    # That helps reduce the network burden of spinning up a development environment
+    image: docker.io/mccutchen/go-httpbin:v2.14.0
+    expose:
+      - "8080"
```
Comment (sarayourfriend) [inline api/compose.yml:64]: I tried a few approaches to testing the trace config, and this was the one that finally worked. Here's what I tried and why they didn't work:

1. I tried pook first, of course. However, the way pook hooks into aiohttp stops it from executing its trace config altogether, so it was useless for testing this.
2. I tried using pytest-httpbin, which runs an httpbin Python server during the pytest suite run, and provides some useful fixtures. However, httpbin itself is very badly out of date and incompatible with other dependencies of the API: https://github.com/postmanlabs/httpbin/issues/712
3. Finally, just running httpbin inside the compose stack, which is what I've done here.

I'm open to suggestions for other approaches that will work and avoid adding this dependency into the stack.

File: api/compose.yml [hunk 2]
```diff
@@ -53,3 +53,12 @@ services:
       DJANGO_NGINX_UPSTREAM_URL: web:50280
     depends_on:
       - web
+
+  httpbin:
```
Comment (sarayourfriend) [inline api/compose.yml:57]:
```suggestion
  # Used in API unit tests
  httpbin:
```

**Review by obulat — APPROVED** [review]
Tested well locally. The thumbnail request time drops sharply when the thumbnail is ready in Photon: from ~1.5 to 0.3.
Is there a plan for monitoring these logs and actions we should take based on the results?

**Comment by sarayourfriend** [comment]
I thought I had created an issue to investigate the extension timings, but I'll do so after merging, and include looking at both timings, not just the extension timings.

**Pull request merged**